server.go

```diff
@@ -1223,48 +1231,91 @@ func (s *Server) ListenAndServeTLSEmbed(addr string, certData, keyData []byte) e
 // ServeTLS serves HTTPS requests from the given listener.
 //
 // certFile and keyFile are paths to TLS certificate and key files.
+//
+// If the certFile or keyFile has not been provided the server structure,
+// the function will use previously added TLS configuration.
 func (s *Server) ServeTLS(ln net.Listener, certFile, keyFile string) error {
-	lnTLS, err := newTLSListener(ln, certFile, keyFile)
-	if err != nil {
+	if err := s.AppendCert(certFile, keyFile); err != nil {
```

Comment:
Now your example program doesn’t work anymore as you are calling this function with an empty certFile and keyFile. You should only call AppendCert here if they aren’t empty strings.